### PR TITLE
use before_action instead of before_filter

### PR DESCRIPTION
**Pull request by Fivell**
refs #4873

### CODE REVIEW

**Comment by Fivell** [comment]
@deivid-rodriguez , no this is a backwards compatible.

before_filter is old rails 3 syntax.  Is alias for before_action in rails 4.0+, 
and it is deprecated in rails 5, and dropped in rails 5.1.   no more reason to have it since we stopped support for rails 3.

**Comment by Fivell** [comment]
@deivid-rodriguez , but you're right we can only change this line https://github.com/activeadmin/activeadmin/pull/4875/commits/248cbf31a96f4f074793c74e9e77e7d39cf45d11#diff-aefa2d67d2ace99bd7a17454959feabdR62

**Comment by deivid-rodriguez** [comment]
Yees, that line is good, we should always use the new syntax internally.

File: lib/active_admin/resource_dsl.rb [hunk 1]
```diff
@@ -180,19 +180,10 @@ def decorate_with(decorator_class)
...
-      actions.each do |action|
-        define_method "#{action}_#{name}" do |*args, &block|
-          controller.public_send "#{action}_action", *args, &block
-        end
-      end
-    end
+    # Standard rails filters
+    delegate :before_action,  :skip_before_action, to: :controller
+    delegate :after_action,   :skip_after_action,  to: :controller
+    delegate :around_action,  :skip_action,        to: :controller
```
Comment (varyonic) [inline lib/active_admin/resource_dsl.rb]: Keep delegating *_filter in DSL for now, right?

Comment (deivid-rodriguez) [inline lib/active_admin/resource_dsl.rb]:
```diff
     delegate :before_save,    :after_save,    to: :controller
     delegate :before_destroy, :after_destroy, to: :controller
 
-    # This code defines both *_filter and *_action for Rails 4.0 to Rails 5.
-    actions = [
+    phases = [
       :before, :skip_before,
       :after,  :skip_after,
       :around, :skip
     ]
-    [:action, :filter].each do |name|
-      actions.each do |action|
+
+    keywords = if Rails::VERSION::MAJOR == 5 && Rails::VERSION::MINOR >= 1
+                 [:action]
+               else
+                 [:action, :filter]
+               end
+
+    keywords.each do |name|
+      phases.each do |action|
         define_method "#{action}_#{name}" do |*args, &block|
           controller.public_send "#{action}_action", *args, &block
         end
```

Comment (Fivell) [inline lib/active_admin/resource_dsl.rb]: @deivid-rodriguez , nice solution!

**Pull request merged**